lib/capistrano-db-tasks/database.rb

```diff
@@ -106,8 +106,13 @@ def dump_cmd_ignore_data_tables_opts
   class Remote < Base
     def initialize(cap_instance)
       super(cap_instance)
-      @config = @cap.capture("cat #{@cap.current_path}/config/database.yml")
-      @config = YAML.load(ERB.new(@config).result)[@cap.fetch(:rails_env).to_s]
+      @cap.within @cap.current_path do
+        @cap.with rails_env: @cap.fetch(:rails_env) do
+          @config = @cap.capture(:rails, 'runner "puts Rails.application.config.database_configuration[Rails.env].to_yaml"', '2>/dev/null')
+          @config.gsub!(/.*\-\-\-/m, '---') # Remove all bundler and rails initialization errors
+        end
```

Comment:
Errors, warnings and notices like:

>    Don't run Bundler as root. Bundler can ask for sudo if it is needed, and installing your bundle as root will break this application for all non-root users on this machine.

or

>    Your Gemfile lists the gem X (>= 0) more than once.
>    You should probably keep only one of them.
>    While it's not a problem now, it could cause errors if you change the version of just one of them later.

which captured with database configuration and breaks the config yaml.